build/circle-deps.sh

```diff
@@ -46,12 +55,25 @@ if [ "${1-}" = "docker" ]; then
     time go build ./gossip/simulation/...
   fi
 
+  exit 0
+fi
+
+shard1_done="${HOME}/shard1"
+shard2_done="${HOME}/shard2"
+
+function notify() {
   if is_shard 2; then
-    time go test -race -v -i ./...
+    time ssh node0 touch "${shard2_done}"
   fi
 
-  exit 0
-fi
+  if is_shard 1; then
+    time ssh node0 touch "${shard1_done}"
+  fi
+}
+
+# Notify shard 0 that shard's 1 and 2 are done if they exit for any
+# reason. This prevents shard 0 from spinning forever.
+trap "notify" 0
```

Comment:
Seems they are equivalent http://unix.stackexchange.com/questions/17314/what-is-signal-0-in-a-trap-command